src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -211,9 +223,16 @@ private void TerminationHookDoneWhenNoChildren()
         protected override void TellInternal(object message, IActorRef sender)
         {
             OnReceive(message);
+            Sender = sender ?? ActorRefs.Nobody;
         }
 
         /// <summary>
+        /// For system messages that require a reply back to the <see cref="IActorRef"/> responsible
+        /// for telling a message to the <see cref="RemoteSystemDaemon"/>.
+        /// </summary>
+        protected IActorRef Sender { get; private set; }
```

Comment:
This is racy.
The `RemoteSystemDaemon` is not an actor, just an impl of `IActorRef` (VirtualPathContainer).
So there is no concurrency constraints applied here.

(I could be misreading something ofc)